src/helpers/validator.js

```diff
@@ -0,0 +1,40 @@
+const _ = require('lodash');
+
+const Symbols = {
+  PERSISTENT: Symbol('persistent'),
+};
+
+module.exports.Symbols = Symbols;
+
+const getValidators = (validators = []) => {
+  if (_.isFunction(validators)) {
+    return [validators];
+  }
+
+  if (!Array.isArray(validators) || !validators.every(_.isFunction)) {
+    throw Error('Validators must be a function or array of functions');
+  }
+
+  return validators;
+};
+
+module.exports.validate = (payload, validators = []) => {
+  const persistentData = payload[Symbols.PERSISTENT];
+  return getValidators(validators).reduce(async (result, validator) => {
+    const data = await result;
```

Comment:
On every iteration we return simple object, not Promise. Why do we need `await` here? And why do we need initial value as a Promise?

Reply:
Promise - agree, it was from my legacy drafts. But we need `await` because reducer's callback is `async` function.